tests/testthat/test-guides.R

```diff
@@ -18,6 +18,30 @@ test_that("Colorbar respects show.legend in layer", {
   expect_true("guide-box" %in% ggplotGrob(p)$layout$name)
 })
 
+test_that("show.legend handles named vectors", {
+  n_legends <- function(p) {
+    g <- ggplotGrob(p)
+    length(g$grobs[[which(g$layout$name == "guide-box")]]) - 1
+  }
+  has_legends <- function(p){
```

Comment:
You mean removing the `has_legends()` test  but keeping the `n_legends()` function the same ?